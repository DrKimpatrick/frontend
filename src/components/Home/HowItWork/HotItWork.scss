.howItWorkSection {
  position: relative;
  background: white;
  .buttons {
    padding: 8px 10px 8px 10px;
    outline: none;
    box-shadow: none;
    font-size: 15px;
    color: rgba(115, 115, 115, 0.2);
    cursor: pointer;
    &:hover{
      color: #737373;
    }
  }
  .active {
    background: #4d9b62;
    color: white !important;
    border-radius: 2px;
  }

  .stepDetail {
    font-size: 14px;
    color: #737373;
    .stepNumber {
      position: relative;
    }
    .description {
      width: 250px;
      margin-top: 1rem;
      min-height: 100px;
    }
  }
}

@media screen and (max-width: 768px) {
  .howItWorkSection {
    .stepDetail {
      li {
        margin-top: 1rem !important;
      }
      .description {
        width: 100%;
        height: auto;
        min-height: auto;
        padding: 0px 10px 0px 10px;
        margin-top: 0.1rem;
      }
    }
  }
}
