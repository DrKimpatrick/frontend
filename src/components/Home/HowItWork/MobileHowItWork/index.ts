import MobileHowItWork from './MobileHowItWork';

export { MobileHowItWork };
