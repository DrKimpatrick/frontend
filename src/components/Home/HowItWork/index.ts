import HowItWork from './HowItWork';

export { HowItWork };
