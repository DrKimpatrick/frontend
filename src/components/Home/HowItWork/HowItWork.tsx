import React, { useState, useEffect } from 'react';
import {
  LaptopMac,
  School as SchoolIcon,
  Work,
  Airplay,
  ImportContacts
} from '@material-ui/icons';
import './HotItWork.scss';
import useWindowSize from 'utils/useWindowSize';
import { MobileHowItWork } from './MobileHowItWork';
import { Step } from './Step';

export enum HowItWorkStep {
  Talent = 'Talent',
  School = 'School',
  Company = 'Companies',
  HumanResource = 'Human Resource',
  Training = 'Training'
}
export const steps = [
  {
    name: HowItWorkStep.Talent,
    icon: <LaptopMac />
  },
  {
    name: HowItWorkStep.School,
    icon: <SchoolIcon />
  },
  {
    name: HowItWorkStep.Training,
    icon: <ImportContacts />
  },
  {
    name: HowItWorkStep.HumanResource,
    icon: <Airplay />
  },
  {
    name: HowItWorkStep.Company,
    icon: <Work />
  }
];
interface Props {
  setStepName: (value: string) => void;
}
const HowItWork = (props: Props) => {
  const [activeStep = HowItWorkStep.Talent, setActiveStep] = useState<string>();

  const { setStepName } = props;

  const size = useWindowSize();

  useEffect(() => {
    setStepName(activeStep);
  }, [activeStep, setStepName]);

  return (
    <div className="howItWorkSection">
      <h2
        data-testid="howItWorks"
        className="flex justify-center mt-16 text-gray-600 font-black text-2xl"
      >
        How it works
      </h2>
      {size && size.width && size.width > 768 ? (
        <div className="my-10">
          <ul className="flex flex-wrap list-none items-center justify-center">
            {steps.map((item, i) => (
              <li
                className="w-full md:w-1/5 flex items-center justify-center"
                key={i}
              >
                <div
                  className={`flex items-center buttons ${
                    activeStep === item.name ? 'active' : ''
                  }`}
                  onClick={() => setActiveStep(item.name)}
                >
                  {item.icon}
                  <h5 className="mx-1">{item.name}</h5>
                </div>
              </li>
            ))}
          </ul>
          <Step activeStep={activeStep} />
        </div>
      ) : (
        <MobileHowItWork
          activeStep={activeStep}
          setActiveStep={setActiveStep}
        />
      )}
    </div>
  );
};

export default HowItWork;
