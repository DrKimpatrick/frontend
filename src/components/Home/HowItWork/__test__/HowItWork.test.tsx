import React from 'react';
import renderer from 'react-test-renderer';
import HowItWork from '../HowItWork';
import { cleanup, render, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';

const mockSetStepName = jest.fn();
describe('HowItWork', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<HowItWork setStepName={mockSetStepName} />, div);
  });

  it('should create snapShot', () => {
    const tree = renderer
      .create(<HowItWork setStepName={mockSetStepName} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should set active step', () => {
    const { container } = render(<HowItWork setStepName={mockSetStepName} />);

    const button: Element | any = container.querySelector('.buttons');

    fireEvent.click(button);

    expect(button).toBeDefined();
  });
});
