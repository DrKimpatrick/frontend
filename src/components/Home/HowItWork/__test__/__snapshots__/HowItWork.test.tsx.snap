// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowItWork should create snapShot 1`] = `
<div
  className="howItWorkSection"
>
  <h2
    className="flex justify-center mt-16 text-gray-600 font-black text-2xl"
    data-testid="howItWorks"
  >
    How it works
  </h2>
  <div
    className="my-10"
  >
    <ul
      className="flex flex-wrap list-none items-center justify-center"
    >
      <li
        className="w-full md:w-1/5 flex items-center justify-center"
      >
        <div
          className="flex items-center buttons active"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 18c1.1 0 1.99-.9 1.99-2L22 5c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2H0c0 1.1.9 2 2 2h20c1.1 0 2-.9 2-2h-4zM4 5h16v11H4V5zm8 14c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
            />
          </svg>
          <h5
            className="mx-1"
          >
            Talent
          </h5>
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center"
      >
        <div
          className="flex items-center buttons "
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
            />
          </svg>
          <h5
            className="mx-1"
          >
            School
          </h5>
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center"
      >
        <div
          className="flex items-center buttons "
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .65.73.45.75.45C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.41.21.75-.19.75-.45V6c-1.49-1.12-3.63-1.5-5.5-1.5zm3.5 14c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
            />
          </svg>
          <h5
            className="mx-1"
          >
            Training
          </h5>
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center"
      >
        <div
          className="flex items-center buttons "
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 22h12l-6-6z"
            />
            <path
              d="M21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h4v-2H3V5h18v12h-4v2h4c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <h5
            className="mx-1"
          >
            Human Resource
          </h5>
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center"
      >
        <div
          className="flex items-center buttons "
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
            />
          </svg>
          <h5
            className="mx-1"
          >
            Companies
          </h5>
        </div>
      </li>
    </ul>
    <ul
      className="flex flex-wrap list-none items-center justify-center my-8 stepDetail"
    >
      <li
        className="w-full md:w-1/5 flex items-center justify-center flex-col text-center"
      >
        <div
          className="font-bold stepNumber"
        >
          01
        </div>
        <div
          className="description"
        >
          Select Talent Account
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center flex-col text-center"
      >
        <div
          className="font-bold stepNumber"
        >
          02
        </div>
        <div
          className="description"
        >
          Confirm your education, employment, and skills level or build your profile from scratch.
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center flex-col text-center"
      >
        <div
          className="font-bold stepNumber"
        >
          03
        </div>
        <div
          className="description"
        >
          Choose your goal Recruiters are 86% more likely to hire pre verified talent.
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center flex-col text-center"
      >
        <div
          className="font-bold stepNumber"
        >
          04
        </div>
        <div
          className="description"
        >
          If you want to be hired choose a premium plan that includes education/employment verification and 3 skill certifications
        </div>
      </li>
      <li
        className="w-full md:w-1/5 flex items-center justify-center flex-col text-center"
      >
        <div
          className="font-bold stepNumber"
        >
          05
        </div>
        <div
          className="description"
        >
          If you want to be hired choose the standard plan that verifies your employment and education.
        </div>
      </li>
    </ul>
  </div>
</div>
`;
